File: app/src/main/java/co/k2/newsbits/data/source/local/LocalDataSource.java
package co.k2.newsbits.data.source.local;

import javax.inject.Inject;

import co.k2.newsbits.data.NewsListener;
import co.k2.newsbits.data.source.DataSource;

/**
 * Copyright (C) 2019 K2 CODEWORKS
 * All rights reserved
 *
 * @author Kislay
 * @since 28/03/19
 */

public class LocalDataSource implements DataSource {

    @Inject
    public LocalDataSource() {

    }

    @Override
    public void getHeadlines(String country, NewsListener listener) {

    }

}

File: app/src/main/java/co/k2/newsbits/data/source/local/LocalDataSource.kt
package co.k2.newsbits.data.source.local

import co.k2.newsbits.data.models.Article
import co.k2.newsbits.data.models.ArticleCacheMetaData

/**
 * Copyright (C) 2019 K2 CODEWORKS
 * All rights reserved
 *
 * @author Kislay
 * @since 28/03/19
 */
interface LocalDataSource {
    suspend fun clearCache()
    suspend fun saveToCache(articles: List<Article>)
    suspend fun getCacheMetaData(): ArticleCacheMetaData?
    suspend fun saveCacheMetaData(articleCacheMetaData: ArticleCacheMetaData)
    suspend fun getHeadlines(country: String?): List<Article>
}
